rails/DEPLOYMENTS.md

```diff
@@ -0,0 +1,51 @@
+
+Abtion Rails deployment environment guidelines:
+
+Draft v0.1
+Authors: 
+	Geoff Hubbard gh@abtion.com
+	Heather Moore hm@abtion.com
+
+Are we all-in on Heroku?
+
+CI:
+Github CI?
+Travis CI?
+Heroku CI
+
+Pipelines for promoting apps from staging to production.
+Automatic deploy from GitHub commits to master branch.
+Preview apps.
+Tests!
+
+Development:
+Local machine.
+Postgres (version 9.x+?)
+
+Staging:
+
+Hobby Dyno - $7
+Heroku Postgres - Hobby Dev - free
+Rollbar - Error logging. free (if your app is generating errors there is a problem fix it.) (upgradable in-place if needed)
+Papertrail - Log logging. free (Upgradable in-place if needed.)
+Librato -  Application Performance Monitoring | Root Cause Analysis. - free
+
+
+Production app:
+
+Professional 1x Dyno - $25
+Heroku Postgres - Standard 0 - $50
+
+Rollbar - free (if your app is generating errors there is a problem fix it.) (upgradable in-place if needed)
+Papertrail - free (Upgradable in-place if needed.)
+Librato - free
+
+
+Needed solutions.
+
+Automatic setup of these environments.
+How to handle exceptions.
+Setup of environment variables.
+Seeding of dev/staging/preview database(s) with "production-like" data.
```

Comment:
The thing is you don't want the exact same data, since you don't want to (for example) be accidentally sending customers testing emails.

Reply:
I think having the same data makes debugging easier. 
You should not be sending emails in development or in staging. For that, we should use [letter opener](https://github.com/ryanb/letter_opener) and [mailtrap](https://mailtrap.io)

Reply:
Depending on the project, we might not want to have the production data accessible to everyone that has access to staging, including consultants and/or other third parties. I think the seeds and fake data makes more sense.
If we decide to pull the data, we should at least make an effort to anonymize it, which would solve the email problem.
I'm not in favor of using catchers on staging, as it would sort of be the same as not executing some code until it reaches production. I'm fine with it in development though.

Reply:
:+1: @prrrnd 
Access to production database should be restricted.